### ISSUE TITLE
stackalloc should be disallowed for structs with ref fields

### ISSUE THREAD

**Issue by jkotas** [dotnet/roslyn#63104]
**Version Used**:   Microsoft.Net.Compilers.Toolset 4.4.0-2.22379.18 (current daily build)

**Steps to Reproduce**:

1. Build 
```csharp
using System;
using System.Reflection;
using System.Runtime.InteropServices;

unsafe
{
    StructWithRefField* p = stackalloc StructWithRefField[10];

    MyClass o = new();    
    // This has GC hole. The stackallocated array of StructWithRefField is not reported to GC
    p[0].RefField = ref o.ByteField;
}

ref struct StructWithRefField
{
    public ref byte RefField;
}

class MyClass
{
    public byte ByteField;
}
```

sharplab is crashing on this repro.

**Expected Behavior**:

Build error. stackalloc should be disallowed for structs with ref fields.

**Actual Behavior**:

Builds successfully, the program has GC hole at runtime.

**Comment by jcouv** [comment]
Assigned to @jaredpar to adjust language rules/spec

**Comment by cston** [comment]
The spec includes the following:
> - A `ref struct` which has a `ref` field is never considered `unmanaged`

This was missed in the implementation.

**Comment by cston** [comment]
Reopening since the change has been reverted - see https://github.com/dotnet/roslyn/pull/63367.

**Comment by MichalPetryka** [comment]
Is this issue maybe a duplicate of #62528?

**Comment by cston** [comment]
@MichalPetryka, this appears to be a distinct issue from #62528, although we should test both scenarios with any potential fix.

**Comment by jcouv** [comment]
Fixed in https://github.com/dotnet/roslyn/pull/64064
